rest/handler.go

```diff
@@ -584,17 +584,33 @@ func (h *handler) writeJSONStatus(status int, value interface{}) {
 		json.Indent(&buffer, jsonOut, "", "  ")
 		jsonOut = append(buffer.Bytes(), '\n')
 	}
+
+	h._writeRawJSONStatus(status, jsonOut)
+}
+
+// writeRawJSONStatus writes the given byte slice as a JSON response.
+// If status is nonzero, the header will be written with that status.
+func (h *handler) writeRawJSONStatus(status int, b []byte) {
+	if !h.requestAccepts("application/json") {
+		base.Warnf(base.KeyAll, "Client won't accept JSON, only %s", h.rq.Header.Get("Accept"))
+		h.writeStatus(http.StatusNotAcceptable, "only application/json available")
+		return
+	}
+	h._writeRawJSONStatus(status, b)
+}
+
+func (h *handler) _writeRawJSONStatus(status int, b []byte) {
```

Comment:
Elsewhere in this repo we mainly use the leading underscore in function names to indicate a locking requirement (i.e. you must hold the mutex before calling this).  I spent a few seconds looking around for a mutex in the callers here - maybe just rename to something

Reply:
Yeah fair comment - I'll think of an appropriate name and update.